services/jupyter/src/notebooks/rubrix/rubrix-active-learning-loop.ipynb

```diff
@@ -0,0 +1,701 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since most query strategies, including ours, require a trained model, we randomly draw a subset from the data pool to initialize our AL system. After obtaining the labels for this batch of instances, the active learner will use them to create the first classifier."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Active Learning loop with Rubrix\n",
+    "\n",
+    "Configure a Rubrix dataset and set up active learning loop"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 126,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# init rubrix\n",
+    "\n",
+    "RUBRIX_URL = os.getenv(\"RUBRIX_API_URL\", \"http://localhost:6900\")\n",
+    "\n",
+    "rb.init(api_url=RUBRIX_URL)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 127,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "settings = rb.TextClassificationSettings(label_schema=LABELS)\n",
+    "# configure dataset\n",
+    "rb.configure_dataset(name=DATASET_NAME, settings=settings)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 131,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 5/5 [00:01<00:00,  4.96it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "5 records logged to http://rubrix:80/datasets/rubrix/bg_test_3\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "BulkResponse(dataset='bg_test_3', processed=5, failed=0)"
+      ]
+     },
+     "execution_count": 131,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# first batch querying with active learner\n",
+    "batch_1_indices = active_learner.query(num_samples=NUM_SAMPLES_ITER)\n",
+    "\n",
+    "records = [\n",
+    "    rb.TextClassificationRecord(\n",
+    "        text=train_text[idx],\n",
+    "        metadata={\"batch_id\": 0},\n",
+    "        id=idx,\n",
+    "    )\n",
+    "    for idx in batch_1_indices\n",
+    "]\n",
+    "\n",
+    "# Log initial records to Rubrix\n",
+    "rb.log(records, DATASET_NAME)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 132,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "LABEL2INT = labeled_dataset[\"train\"].features[\"label\"].str2int"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 133,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ACCURACIES = []"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 134,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@listener(dataset=DATASET_NAME,\n",
```

Comment:
I'm a bit confused how the @listener interacts with active_learning_loop and would prefer more glassbox solution, but I guess this is fine due to time constraints.

Reply:
Yeah  I agree but given the time constraints... I don't see any easy alternative I'm afraid